EuclideanGeometry/Foundation/Construction/Triangle/IsocelesTriangle.lean

```diff
@@ -12,15 +25,56 @@ variable {P : Type _} [EuclideanPlane P]
 
 section IsocelesTriangles
 
+open Triangle
 def Triangle.IsIsoceles (tri : Triangle P) : Prop := tri.edge₂.length = tri.edge₃.length
 
-theorem is_isoceles_tri_iff_ang_eq_ang_of_nd_tri {tri_nd : Triangle_nd P} : tri_nd.1.IsIsoceles ↔ (tri_nd.angle₂.value= - tri_nd.angle₃.value) := sorry
+
+
+
+theorem is_isoceles_tri_iff_ang_eq_ang_of_nd_tri {tri_nd : Triangle_nd P} : tri_nd.1.IsIsoceles ↔ (tri_nd.angle₂.value=  tri_nd.angle₃.value) := by
 -- To show angle equal => sides equal, use anti-congruent of the triangle with itself. SAS
+rcases Triangle.edge_eq_edge_of_flip_vertices tri_nd.1 with ⟨a, b, c⟩
+rcases Triangle.angle_eq_neg_angle_of_flip_vertices tri_nd with ⟨x, y, z⟩ 
+have h21 :  tri_nd.1.edge₃.length = tri_nd.flip_vertices.1.edge₂.length := c
+have h22 : tri_nd.1.edge₂.length = tri_nd.flip_vertices.1.edge₃.length := b
+have h23 : tri_nd.angle₁.value = - tri_nd.flip_vertices.angle₁.value := x
+have h24 : tri_nd.angle₃.value = - tri_nd.flip_vertices.angle₂.value := z
+
+constructor
+intro h0
+have h1 : tri_nd.1.edge₂.length = tri_nd.1.edge₃.length := by
+  exact h0  
+have _ : tri_nd.1 IsACongrTo tri_nd.flip_vertices.1 := by
+  apply acongr_of_SAS
+  rw[← h21 , h1]
```

Comment:
You can try to use `rw [← Triangle.edge_eq_edge_of_flip_vertices tri_nd.1]` or `.2.1, .2.2` instead of use `have`'s.